stdlib/LinearAlgebra/src/symmetric.jl

```diff
@@ -920,7 +920,7 @@ function ^(A::Hermitian{T}, p::Real) where T
     end
 end
 
-for func in (:exp, :cos, :sin, :tan, :cosh, :sinh, :tanh, :atan, :asinh, :atanh)
+for func in (:exp, :cis, :cos, :sin, :tan, :cosh, :sinh, :tanh, :atan, :asinh, :atanh)
```

Comment:
I am pretty sure this is not valid, since `cis` can produce complex values for real inputs, but symmetric/Hermitian matrices can only have real eigenvalues. An example of `cis` producing a matrix that is non-Hermitian for symmetric/Hermitian inputs would be the following:
```julia
julia> exp(im * [π 0; 0 π] / 2)
2×2 Matrix{ComplexF64}:
 2.22045e-16+1.0im          0.0+0.0im
         0.0+0.0im  2.22045e-16+1.0im
```
We'd therefore need to drop the `Symmetric`/`Hermitian` wrappers here. (The output should always be unitary, but AFAIK, that's not something we are currently able to take advantage of in LinearAlgebra)

Reply:
Right, `Hermitian` should be removed, but the resulting matrix will always be `Symmetric` (with complex diagonals, eventually). Should we keep `Symmetric`?

Reply:
You are right `cispi(::Symmetric{<:Real})` should always produce a complex symmetric matrix. Complex symmetric matrices might be a little odd though and I am not sure we take advantage of their properties anywhere in LinearAlgebra, so it might not be worth bothering.

Reply:
OK, I'll leave it unwrapped.

Reply:
@simeonschaub I just realized `Symmetric{Complex{T},S}` is used in some places in `LinearAlgebra` (in particular because `RealHermSymComplexSym` is a `Union` that includes it), and there are even specific methods for it in `symmetric.jl`. Should we really not wrap `cis` in this case?

Reply:
I might not be the best person to answer that, maybe @dkarrasch or @stevengj can comment on that

Reply:
I wouldn't put it in `Symmetric`, which will mainly just run the danger of breaking code (due to missing methods for `Symmetric`) or being slow (because of missing BLAS support), for not much benefit.

Ff you want to take advantage of any special property of `cis(hermitian)`, it would be that the matrix is unitary.  Being complex-symmetric (in the real-symmetric case) is mostly irrelevant, because complex-symmetric matrices simply aren't very special (*every* matrix is similar to a complex-symmetric matrix).

Reply:
Perfect, removed!